### PR TITLE
Added 'between' operator for nodes and case insensitive parsers

### PR DESCRIPTION
**Pull request by JuankS92**
#### What does this PR do?

Simple enhancements to utils/Query so it understands operator 'between' and node values can be lowercase.
#### Where should the reviewer start?

utils/Query
#### What testing has been done on this PR?

Is being used in our current project. Tests added on this branch: N/A.
#### How should this be manually tested?

Provide Query a simple text to parse. Include operator 'between'.
`new Query('(BETWEEN x y) AND (n:v OR n:w) "big deal" NOT 01:23:45:67:89:ab 2016-01-31T16:45:46Z');`
#### Any background context you want to provide?

Our current project uses Query. Operators like 'between' can be simply added to the parsers list. Other extra enhancements are just an improvement over small limitations, like parsing node values only on 
capital letters.
#### What are the relevant issues?

None yet.
#### Screenshots (if appropriate)

N/A.
#### Do the grommet-index docs need to be updated?

Absolutely necessary.
#### Should this PR be mentioned in the release notes?

Yes.
#### Is this change backwards compatible or is it a breaking change?

Yes - backwards compatible.

### CODE REVIEW

File: src/js/utils/Query.js [hunk 1]
```diff
@@ -3,16 +3,15 @@
...
+// This pattern matches the name: ^[^\d='"\s]{2}[^='"\s]+
 // We allow for the value to be optionally be quoted. So, we repeat the name
 // pattern three times, once for single quoted value, once for double quoted
 // value, and lastly with no quotes.
 // We don't build this programmatically for better performance.
 const ATTRIBUTE_PATTERN = new RegExp([
-  `^[^\d:'"\s]{1}[^:'"\s]*:'[^']+'`,
-  `^[^\d:'"\s]{1}[^:'"\s]*:"[^"]+"`,
-  `^[^\d:'"\s]{1}[^:'"\s]*:[^'"\s]+`].join('|'));
+  `^[^\d='"\s]{1}[^='"\s]*='[^']+'`,
```
Comment (ericsoderberghp) [inline src/js/utils/Query.js]: Why did you change from `:` as the separator between name and value to `=`?

**Review by ericsoderberghp — COMMENTED** [review]
I like the BETWEEN and case insensitive changes. One question about the name/value separator character though. That seems like a breaking change for products like HC380 that already use `:`.

File: src/js/utils/Query.js (hunk 1 above)
Comment (JuankS92) [reply src/js/utils/Query.js]: The attribute:value parser has some limitations. The only reason I added that is because we need to parse dates and timestamps. Also I believe is a more 'sql like' value to compare. I think it's not completely necessary to add it if you want it not to be a breaking change.

Comment (ericsoderberghp) [reply src/js/utils/Query.js]: The original rationale for `:` was to align with internet search syntax conventions. Both Google and Bing use `:`. For example: `filetype:mp3` or `site:grommet.io`. I'd prefer to split this aspect of this pull request out for now. I would like to consider making the separator configurable via a property. I'm trying to weigh the trade-offs between having app specific separators that could be more domain appropriate vs. having consistency across apps.

Comment (JuankS92) [reply src/js/utils/Query.js]: Interesting. I restored the original attribute:value parser. It's not a big deal right now because the values we have can be parsed in the original way. I know that in the future there will be more complex data to deal with, so an app-specific separator might be just what we need.
